dotnet/src/SemanticKernel.Core/Data/TextSearch/TextSearchServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.SemanticKernel.Data;
+using Microsoft.SemanticKernel.Embeddings;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Extension methods to register <see cref="ITextSearch"/> for use with <see cref="IServiceCollection"/>.
+/// </summary>
+public static class TextSearchServiceCollectionExtensions
+{
+    /// <summary>
+    /// Register an <see cref="ITextSearch"/> instance with the specified service ID.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="ITextSearch"/> on.</param>
+    /// <param name="textSearch">Instance of <see cref="ITextSearch"/> to register.</param>
+    /// <param name="serviceId">An optional service id to use as the service key.</param>
+    public static IServiceCollection AddTextSearch(
+        this IServiceCollection services,
+        ITextSearch textSearch,
+        string? serviceId = default)
+    {
+        ITextSearch Factory(IServiceProvider serviceProvider, object? _) => textSearch;
+        services.AddKeyedSingleton<ITextSearch>(serviceId, (Func<IServiceProvider, object?, ITextSearch>)Factory);
+        return services;
+    }
+
+    /// <summary>
+    /// Register an <see cref="IVectorizableTextSearch{TRecord}"/> instance with the specified service ID.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="ITextSearch"/> on.</param>
+    /// <param name="vectorTextSearch">Instance of <see cref="IVectorizableTextSearch{TRecord}"/> to register.</param>
+    /// <param name="serviceId">An optional service id to use as the service key.</param>
+    public static IServiceCollection AddVectorStoreTextSearch<TRecord>(
+        this IServiceCollection services,
+        VectorStoreTextSearch<TRecord> vectorTextSearch,
+        string? serviceId = default)
+        where TRecord : class
+    {
+        VectorStoreTextSearch<TRecord> Factory(IServiceProvider serviceProvider, object? _) => vectorTextSearch;
+        services.AddKeyedSingleton<VectorStoreTextSearch<TRecord>>(serviceId, (Func<IServiceProvider, object?, VectorStoreTextSearch<TRecord>>)Factory);
+        return services;
+    }
+
+    /// <summary>
+    /// Register a <see cref="VectorStoreTextSearch{TRecord}"/> instance with the specified service ID.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="ITextSearch"/> on.</param>
+    /// <param name="serviceId">An optional service id to use as the service key.</param>
+    public static IServiceCollection AddVectorStoreTextSearch<TRecord>(
```

Comment:
Consider an overload where we take the mappers and options as input, but resolve the `IVectorizableTextSearch` from DI.  I would expect that to be a common scenario, e.g. the user is using the existing helpers for VectorStore to register an instance of IVectorizableTextSearch, and now they want to consume that and provide some specific options for how to consume it.